File: python/1461B.py


t = int(input())

trees = []
for i in range(500):
    trees.append([0] * 500)

for _ in range(t):
    n, m = map(int, input().split())
    counter = 0

    for i in range(n):
        s = input()
        for j in range(m):
            if (s[j] == '*'):
                trees[i][j] = 1
                counter += 1
            else:
                trees[i][j] = 0

    for required_size in range(1, min(n, m)+1):
        for i in range(required_size, n):
            for j in range(required_size, m - required_size):
                if trees[i][j] >= 1:
                    if trees[i][j-1] >= required_size and trees[i-1][j] >= required_size and trees[i][j+1] >= required_size:
                        counter += 1
                        trees[i][j] = required_size + 1

    print(counter)

File: cpp/1461B.cpp
#include <iostream>
#include <vector>
#include <algorithm>

using namespace std;
using ll = long long;

int main(){

    cin.tie(0);
    cout.tie(0);
    ios_base::sync_with_stdio(0);

    int t;
    cin >> t;

    vector<vector<int>> trees(500, vector<int>(500));

    for (int q = 0; q < t; q++){
        int n, m, counter = 0;
        cin >> n >> m;

        char el;
        for (int i = 0; i < n; i++)
            for (int j = 0; j < m; j++){
                cin >> el;
                if (el == '*'){
                    trees[i][j] = 1;
                    counter++;
                }
                else{
                    trees[i][j] = 0;
                }

            }

        for (int required_size = 1; required_size <= min(n, m); required_size++){
            for (int i = required_size; i < n; i++){
                for (int j = required_size; j < m - required_size; j++){
                    if (trees[i][j] >= 1){
                        if (trees[i-1][j] >= required_size &&
                            trees[i][j-1] >= required_size &&
                            trees[i][j+1] >= required_size)
                        {
                            
                            counter++;
                            trees[i][j] = required_size + 1;
                        }
                    }
                }
            }
        }
        
        cout << counter << '\n';
        
    }

    return 0;
}